server/src/test/java/io/crate/integrationtests/GeoShapeIntegrationTest.java

```diff
@@ -192,4 +193,12 @@ public void testGeoPointInPolygonQueryLuceneBug() {
                 "-0.129089 51.536726))')");
         assertThat(response.rowCount(), is(0L));
     }
+
+    @Test
+    public void test_can_add_geo_shape_array_column() throws Exception {
+        execute("create table t(id int)");
+        execute("alter table t add column shapes array(geo_shape)");
+        execute("insert into t(id, shapes) VALUES (?, ?)", $(3, new Object[]{GEO_SHAPE1, GEO_SHAPE2}));
+        Assertions.assertThat(response.rowCount()).isEqualTo(1L);
+    }
```

Comment:
This passes on master and fails on 5.2 with error:

without cherry-picking the reference commit fails with 

`Cannot cast value [{type=LineString, coordinates=[[D@9d5bc89}, {type=LineString, coordinates=[[D@23bc15b4} to type geo_shape` since type is lost, array becomes plain.